chore: Add Prettier

baseballyama opened the pull request:
I don't think we will merge this PR now. But just I configured Prettier.
After merging all Svelte4 PRs, I will rebase this PR and I will re-run Prettier.

### Before submitting the PR, please make sure you do the following
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] Prefix your PR title with `feat:`, `fix:`, `chore:`, or `docs:`.
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
-  [x] Run the tests with `npm test` and lint the project with `npm run lint`

benmccann reviewed (approved):
woohoo!

dummdidumm commented:
Couple of notes:
- We should add `"printWidth": 100` to the config so lines don't break as often. We have the same setting in the SvelteKit repo
- We should not format `site` right now. It's being heavily rewritten in a branch and that would become a merge nightmare

jamesscottbrown commented:
After rebasing, I suggest creating a `.git-blame-ignore-revs` file and adding the commits that apply the prettier changes, so that `git blame` shows the most recent meaningful (rather than formatting) changes: see [Ignore commits in the blame view](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view)

benmccann commented:
@baseballyama it looks like pretty much all the Svelte 4 PRs have been merged. I can take care of rebasing the last couple after this is merged if you'd like to go ahead and update this

baseballyama commented:
> We should add "printWidth": 100 to the config so lines don't break as often. We have the same setting in the SvelteKit repo

OK!

> We should not format site right now. It's being heavily rewritten in a branch and that would become a merge nightmare

Yeah, I think we can merge this PR before site one. So I will ignore site dir.

> After rebasing, I suggest creating a .git-blame-ignore-revs file and adding the commits that apply the prettier changes, so that git blame shows the most recent meaningful (rather than formatting) changes: see [Ignore commits in the blame view](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view)

I faced few behavior changed while working on prettier. (And I adjusted the configuration.)
So it is better to be able to blame in case a bug occurs.

> it looks like pretty much all the Svelte 4 PRs have been merged. I can take care of rebasing the last couple after this is merged if you'd like to go ahead and update this

Oh is it.
I remember agreeing to convert TS to JS + JSDoc and switch test runner.
Does this mean that these tasks can be done after formatting? (I can say yes but I would like to make a confirmation)

baseballyama commented:
rebased!

dominikg commented:
it would be great to add .editorconfig, .gitattributes and the prettierrc from sveltekit so that the formatting is similar

baseballyama commented:
> it would be great to add .editorconfig, .gitattributes 

Oh! I will add this!

> the prettierrc from sveltekit so that the formatting is similar

OK. but why kit repo doesn't use `prettier-plugin-svelte`?

baseballyama commented:
> it would be great to add .editorconfig, .gitattributes

Already both files exist and configuration is almost same.

> the prettierrc from sveltekit so that the formatting is similar

I updated.
But I removed `requirePragma` config. Because it's difficult to add `@format` comment proactive. (In fact, no one use `@format` in kit repo.)

dummdidumm reviewed (approved):
Lessgoooo
